src/librustc_mir/interpret/eval_context.rs

```diff
@@ -556,34 +570,40 @@ impl<'a, 'mir, 'tcx: 'mir, M: Machine<'a, 'mir, 'tcx>> InterpretCx<'a, 'mir, 'tc
         for local in frame.locals {
             self.deallocate_local(local.value)?;
         }
-        // Validate the return value. Do this after deallocating so that we catch dangling
-        // references.
-        if let Some(return_place) = frame.return_place {
-            if M::enforce_validity(self) {
-                // Data got changed, better make sure it matches the type!
-                // It is still possible that the return place held invalid data while
-                // the function is running, but that's okay because nobody could have
-                // accessed that same data from the "outside" to observe any broken
-                // invariant -- that is, unless a function somehow has a ptr to
-                // its return place... but the way MIR is currently generated, the
-                // return place is always a local and then this cannot happen.
-                self.validate_operand(
-                    self.place_to_op(return_place)?,
-                    vec![],
-                    None,
-                    /*const_mode*/false,
-                )?;
+
+        // When we're popping a stack frame for unwinding purposes,
+        // we don't care at all about returning-related stuff (i.e. return_place
+        // and return_to_block), because we're not performing a return from this frame.
+        if !unwinding {
```

Comment:
move the body of this `if` to `pop_stack_frame` after the call to `pop_stack_frame_internal`

Reply:
Addressed